src/libs/model/index.ts

```diff
@@ -0,0 +1,213 @@
+      this.injectFunctions(element);
+    });
+
+    // Resolving references
+    // console.log('Resolving references.');
+    this.getElements().forEach((element: IElement) => {
+      this.replaceReferences(element);
+    });
+  }
+
+  /**
+   * Function that returns an array of contained IElement objects and updates the container IReference of all elements along the way.
+   */
+  getElements(current?: IPackage): IElement[] {
+    let contents: IElement[];
+
+    if (!current) {
+      current = this.rootPackage;
+      contents = [current, ...(current.contents ? current.contents : [])];
+    } else {
+      contents = [...(current.contents ? current.contents : [])];
+    }
+
+    current.container = null;
+
+    if (!current.contents) {
+      return [];
+    }
+
+    current.contents.forEach((element: IElement) => {
+      if (element.type === OntoUMLType.PACKAGE_TYPE) {
+        contents = [...contents, ...this.getElements(element as IPackage)];
+      } else if (element.type === OntoUMLType.RELATION_TYPE) {
+        const relation = element as IRelation;
+
+        if (relation.properties) {
+          relation.properties.forEach(
+            property => (property.container = relation),
+          );
+        }
+        contents = [
+          ...contents,
+          ...(relation.properties ? relation.properties : []),
+        ];
+      } else if (element.type === OntoUMLType.CLASS_TYPE) {
+        const _class = element as IClass;
+
+        if (_class.properties) {
+          _class.properties.forEach(property => (property.container = _class));
+        }
+        if (_class.literals) {
+          _class.literals.forEach(literal => (literal.container = _class));
+        }
+        contents = [
+          ...contents,
+          ...(_class.properties ? _class.properties : []),
+          ...(_class.literals ? _class.literals : []),
+        ];
+      }
+
+      element.container = { type: current.type, id: current.id };
+    });
+
+    return contents;
+  }
+
+  /**
+   * Returns an IElement according to its id.
+   *
+   * @param elementId - Desired element's id.
+   */
+  getElementById(elementId: string): any {
+    return this.getElements().find(element => element.id === elementId);
+  }
+
+  checkInstanceOfIContainer(element: IElement): boolean {
+    return [
+      OntoUMLType.PACKAGE_TYPE,
+      OntoUMLType.CLASS_TYPE,
+      OntoUMLType.RELATION_TYPE,
+    ].includes(element.type);
+  }
+
+  checkInstanceOfIDecoratable(element: IElement): boolean {
+    return [
+      OntoUMLType.PROPERTY_TYPE,
+      OntoUMLType.CLASS_TYPE,
+      OntoUMLType.RELATION_TYPE,
+    ].includes(element.type);
+  }
+
+  checkInstanceOfIClassifier(element: IElement): boolean {
+    return [OntoUMLType.CLASS_TYPE, OntoUMLType.RELATION_TYPE].includes(
+      element.type,
+    );
+  }
+
+  injectFunctions(element: IElement, enableMemoization = true): void {
+    Object.keys(functions.IElement_functions).forEach(
```

Comment:
You can put this piece of code in another function passing the functions as parameter:

```
Object.keys(functions.IElement_functions).forEach(
      (functionName: string) => {
        element[functionName] = enableMemoization
          ? memoizee(functions.IElement_functions[functionName])
          : functions.IElement_functions[functionName];
      },
    );
```

Example:

`injectElementMethods(functions.IElement_functions)`